uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/LeaveRequestTest.php

```diff
@@ -1312,6 +1312,406 @@ public function testGetFullReturnsOnlyTheOriginalAmountWhenTheExpiredParamIsNotP
+      'contact_id' => $staffMember2['id'],
+      'type_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('2016-01-05'),
+      'to_date' => CRM_Utils_Date::processDate('2016-01-05'),
+      'from_date_type' => 1,
+      'to_date_type' => 1
+    ], true);
+
+
+    // Only staffMember1 is actively managed by the Leave Manager
+    // therefore only leave requests for staffMember1 is returned
+    $result = civicrm_api3('LeaveRequest', 'get', ['unassigned' => false, 'check_permissions' => true]);
+    $resultGetFull = civicrm_api3('LeaveRequest', 'getFull', ['unassigned' => false, 'check_permissions' => true]);
+
+    $this->assertEquals(1, $result['count']);
+    $this->assertNotEmpty($result['values'][$leaveRequest1->id]);
+
+    $this->assertEquals(1, $resultGetFull['count']);
+    $this->assertNotEmpty($resultGetFull['values'][$leaveRequest1->id]);
+  }
+
+  public function testGetAndGetFullShouldReturnEmptyResponseWhenManagedByParameterIsPresentAndUnassignedIsTrue() {
+    $this->setLeaveApproverRelationshipTypes([
+      'has Leaves Approved By',
+      'has Leaves Managed By',
+    ]);
+
+    $manager1 = ContactFabricator::fabricate();
+
+    $staffMember1 = ContactFabricator::fabricate();
+    $staffMember2 = ContactFabricator::fabricate();
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $staffMember1['id']],
+      ['period_start_date' => '2016-01-01']
+    );
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $staffMember2['id']],
+      ['period_start_date' => '2015-10-01']
+    );
+
+    // Set Leave Approvers for staffMembers 1 and 2.
+    // staffMember2 does not have an active leave manager relationship
+    $this->setContactAsLeaveApproverOf($manager1, $staffMember1, null, null, true, 'has Leaves Approved By');
+    $this->setContactAsLeaveApproverOf($manager1, $staffMember2, '2016-01-01', '2016-12-28', true, 'has Leaves Managed By');
+
+    $leaveRequest1 = LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $staffMember1['id'],
+      'type_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('2016-01-01'),
+      'to_date' => CRM_Utils_Date::processDate('2016-01-01'),
+      'from_date_type' => 1,
+      'to_date_type' => 1
+    ], true);
+
+    $leaveRequest2 = LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $staffMember2['id'],
+      'type_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('2016-01-05'),
+      'to_date' => CRM_Utils_Date::processDate('2016-01-05'),
+      'from_date_type' => 1,
+      'to_date_type' => 1
+    ], true);
+
+    //No result is returned because the values on both managed_by and unassigned parameters contradict
+    //i.e you cannot get results for contacts with active managers with unassigned true at the site time
+    //its a contradiction.
+    $result = civicrm_api3('LeaveRequest', 'get', ['managed_by' => $manager1['id'], 'unassigned' => true]);
```

Comment:
for this test, I think only one contact, contract and leave request is enough, since no matter the number of records, this combination of params will always result in an empty result set